packages/manager/modules/dbaas-logs/src/logs/detail/streams/add/logs-streams-add.controller.js

```diff
@@ -42,91 +42,63 @@ export default class LogsStreamsAddCtrl {
-            add.plan.planCode ===
-            this.LogsConstants.CONSUMPTION_REFERENCE.COLDSTORAGE_PCA,
-        );
-        const coldstoragePCSCapacities = selectedFamily.addons.find(
-          (add) =>
-            add.plan.planCode ===
-            this.LogsConstants.CONSUMPTION_REFERENCE.COLDSTORAGE_PCS,
-        );
-        const indexingFirstStepPrice = indexingCapacities.plan.details.pricings.default.find(
-          (capabilities) =>
-            capabilities.capacities.includes(
-              this.LogsConstants.CONSUMPTION_CAPACITY,
-            ) &&
-            capabilities.maximumQuantity ===
-              this.LogsConstants.INDEXING_TIERING,
-        );
-        const indexingSecondStepPrice = indexingCapacities.plan.details.pricings.default.find(
-          (capabilities) =>
-            capabilities.capacities.includes(
-              this.LogsConstants.CONSUMPTION_CAPACITY,
-            ) &&
-            capabilities.minimumQuantity ===
-              this.LogsConstants.INDEXING_TIERING + 1,
-        );
-        const coldstoragePCA = coldstoragePCACapacities.plan.details.pricings.default.find(
-          (capabilities) =>
-            capabilities.capacities.includes(
-              this.LogsConstants.CONSUMPTION_CAPACITY,
-            ),
-        );
-        const coldstoragePCS = coldstoragePCSCapacities.plan.details.pricings.default.find(
-          (capabilities) =>
-            capabilities.capacities.includes(
-              this.LogsConstants.CONSUMPTION_CAPACITY,
-            ),
-        );
-        this.coldStoragePrice.PCA.price = coldstoragePCA.price.text;
-        this.coldStoragePrice.PCS.price = coldstoragePCS.price.text;
-        this.indexingStoragePrice.FirstStep.price =
-          indexingFirstStepPrice.price.text;
-        this.indexingStoragePrice.SecondStep.price =
-          indexingSecondStepPrice.price.text;
-      });
-    this.encryptionKeys = this.CucControllerHelper.request.getHashLoader({
-      loaderFunction: () =>
-        this.LogsEncryptionKeysService.getEncryptionKeys(this.serviceName),
+  $onInit() {
+    this.catalog.load().then((catalog) => {
+      const selectedCatalog = catalog.plans.find(
+        (plan) => plan.planCode === this.LogsConstants.LDP_PLAN_CODE,
+      );
+      const selectedFamily = selectedCatalog.addonsFamily.find(
+        (addon) => addon.family === this.LogsConstants.ADD_ON_FAMILY.NEW,
+      );
+      const indexingCapacities = selectedFamily.addons.find(
+        (add) =>
+          add.plan.planCode === this.LogsConstants.CONSUMPTION_REFERENCE.STREAM,
+      );
+      const coldstoragePCACapacities = selectedFamily.addons.find(
+        (add) =>
+          add.plan.planCode ===
+          this.LogsConstants.CONSUMPTION_REFERENCE.COLDSTORAGE_PCA,
+      );
+      const coldstoragePCSCapacities = selectedFamily.addons.find(
+        (add) =>
+          add.plan.planCode ===
+          this.LogsConstants.CONSUMPTION_REFERENCE.COLDSTORAGE_PCS,
+      );
```

Comment:
**nitpick(perf):** Since `selectedFamily` array is looped thrice, it can be avoided and iterated once to find the required info from the single iteration.